### PR TITLE
[OPIK-4140] [FE] [BE] improve version selection UX

### PR DESCRIPTION
**Pull request by awkoy**
## Details

Backend changes:
- Added versionCount field to Dataset API response
- Implemented DatasetVersionCount record and DAO method to fetch version counts
- Updated DatasetService to populate version counts when fetching datasets

Frontend changes:
- Refactored DatasetVersionSelectBox to improve user experience
- Clicking dataset now selects latest version directly
- Version dropdown chevron only shown when multiple versions exist
- Improved layout with better spacing and hover states
- Fixed version selection state management with debounced close logic

## Change checklist
- [ ] User facing
- [ ] Documentation update

## Issues

- Resolves #
- OPIK-4140

### CODE REVIEW

**Comment by CometActions** [comment]
❌ **Test environment deployment failed**

The deployment encountered an error. Please check the [deployment logs](https://github.com/comet-ml/comet-deployment/actions/runs/21590099743) for details.

**Comment by CometActions** [comment]
❌ **Test environment deployment failed**

The deployment encountered an error. Please check the [deployment logs](https://github.com/comet-ml/comet-deployment/actions/runs/21590099743) for details.

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-5008.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-5008
- **Version:** 1.10.2-5008-merge-1126
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-5008%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-5008.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-5008
- **Version:** 1.10.2-5008-merge-1133
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-5008%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-5008.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-5008
- **Version:** 1.10.2-5008-merge-1135
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-5008%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java [hunk 1]
```diff
@@ -207,7 +207,7 @@ class DatasetsResourceTest {
     public static final String[] DATASET_IGNORED_FIELDS = {"id", "createdAt", "lastUpdatedAt", "createdBy",
             "lastUpdatedBy", "experimentCount", "mostRecentExperimentAt", "lastCreatedExperimentAt",
             "datasetItemsCount", "lastCreatedOptimizationAt", "mostRecentOptimizationAt", "optimizationCount",
-            "status", "latestVersion"};
+            "versionCount", "status", "latestVersion"};
```
Comment (thiagohora) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java]: Add at least one test to confirm that the count returns the expected value

Comment (thiagohora) [reply apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java]: Also test what happens if the dataset list is empty

**Review by andriidudar — DISMISSED** [review]
The FE part looks good. Nice work — approved!

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-5008.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-5008
- **Version:** 1.10.2-5008-merge-1145
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-5008%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

**To configure additional Environment** variables for your environment, run [Deploy Opik AdHoc Environment workflow] (https://github.com/comet-ml/comet-deployment/actions/workflows/deploy_opik_adhoc_env.yaml)

### Access Information
- **URL:** https://pr-5008.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-5008
- **Version:** 1.10.2-5008-merge-1149
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-5008%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.
